fdbserver/RestoreRoleCommon.actor.h

```diff
@@ -0,0 +1,138 @@
+/*
+ * RestoreRoleCommon.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This file delcares common struct and functions shared by restore roles, i.e.,
+// RestoreMaster, RestoreLoader, RestoreApplier
+
+#pragma once
+#if defined(NO_INTELLISENSE) && !defined(FDBSERVER_RestoreRoleCommon_G_H)
+#define FDBSERVER_RestoreRoleCommon_G_H
+#include "fdbserver/RestoreRoleCommon.actor.g.h"
+#elif !defined(FDBSERVER_RestoreRoleCommon_H)
+#define FDBSERVER_RestoreRoleCommon_H
+
+#include <sstream>
+#include "flow/Stats.h"
+#include "fdbclient/FDBTypes.h"
+#include "fdbclient/CommitTransaction.h"
+#include "fdbrpc/fdbrpc.h"
+#include "fdbrpc/Locality.h"
+#include "fdbserver/CoordinationInterface.h"
+#include "fdbserver/RestoreWorkerInterface.h"
+#include "fdbserver/RestoreUtil.h"
+
+#include "flow/actorcompiler.h" // has to be last include
+
+extern bool debug_verbose;
+
+struct RestoreRoleInterface;
+struct RestoreLoaderInterface;
+struct RestoreApplierInterface;
+
+struct RestoreRoleData;
+struct RestoreMasterData;
+
+struct RestoreSimpleRequest;
+
+typedef std::map<Version, Standalone<VectorRef<MutationRef>>> VersionedMutationsMap;
+
+ACTOR Future<Void> handleHeartbeat(RestoreSimpleRequest req, UID id);
+ACTOR Future<Void> handleInitVersionBatchRequest(RestoreVersionBatchRequest req, Reference<RestoreRoleData> self);
+ACTOR Future<Void> handleFinishRestoreRequest(RestoreVersionBatchRequest req, Reference<RestoreRoleData> self);
+
+// Helper class for reading restore data from a buffer and throwing the right errors.
+// This struct is mostly copied from StringRefReader. We add a sanity check in this struct.
+// TODO: Merge this struct with StringRefReader.
```

Comment:
Were you planning on doing this now?

Reply:
Nope, the merging between this restore and existing restore comes after this PR gets merged.
I want to reduce the amount of work done in this PR.